resilience4j-rxjava2/src/main/java/io/github/resilience4j/bulkhead/operator/AbstractBulkheadOperator.java

```diff
@@ -0,0 +1,46 @@
+package io.github.resilience4j.bulkhead.operator;
+
+import io.github.resilience4j.bulkhead.Bulkhead;
+import io.github.resilience4j.bulkhead.BulkheadFullException;
+import io.github.resilience4j.internal.PermittedOperator;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A disposable bulkhead acting as a base class for bulkhead operators.
+ *
+ * @param <T> the type of the emitted event
+ * @param <D> the actual type of the disposable/subscription
+ */
+abstract class AbstractBulkheadOperator<T, D> extends PermittedOperator<T, D> {
+    private final Bulkhead bulkhead;
```

Comment:
make bulkhead transient or seriazlizable as Fields in a Serializable class must themselves be either Serializable or transient even if the class is never explicitly serialized or deserialized. For instance, under load, most J2EE application frameworks flush objects to disk, and an allegedly Serializable object with non-transient, non-serializable data members could cause program crashes, and open the door to attackers. In general a Serializable class is expected to fulfil its contract and not have an unexpected behaviour when an instance is serialized.
This rule raises an issue on non-Serializable fields, and on collection fields when they are not private (because they could be assigned non-Serializable values externally), and when they are assigned non-Serializable types within the class.

Reply:
Sorry for not getting back sooner. 

Not sure what should be done here..
I don't think making it transient would make any sense. It would break anyway...
Should I not extend an `AtomicReference` but rather move it to a field?

Reply:
Yes i guess it would be better